src/index.js

```diff
@@ -1,10 +1,223 @@
+ * @throws {Error} If attempting to use embedOptions.responsiveIframe with a
+ *                 non-iframe embed.
+ * @throws {Error} If refNodeInsert is missing or invalid.
+ */
+const checkParams = (params) => {
+  const {
+    accountId,
+    embedOptions,
+    embedType,
+    options,
+    refNode,
+    refNodeInsert
+  } = params;
+
+  let err;
+
+  if (!accountId) {
+    err = 'accountId is required';
+  } else if (!refNode) {
+    err = 'refNode is required';
+  } else if (typeof refNode !== 'string' &&
+             (refNode.nodeType !== 1 || !refNode.parentNode)) {
+    err = 'if refNode is not a string, it must be a DOM node with a parent';
+  } else if (!isValidEmbedType(embedType)) {
+    err = 'embedType is missing or invalid';
+  } else if (embedType === EMBED_TYPE_IFRAME && options) {
+    err = 'cannot use options with an iframe embed';
+  } else if (embedOptions &&
+             embedOptions.responsive &&
+             embedOptions.responsive.aspectRatio &&
+             !(/^\d+\:\d+$/).test(embedOptions.responsive.aspectRatio)) {
+    err = 'embedOptions.responsive.aspectRatio must be in the "n:n" format';
+  } else if (!isValidRootInsert(refNodeInsert)) {
+    err = 'refNodeInsert is missing or invalid';
+  }
+
+  if (err) {
+    throw new Error(err);
+  }
+};
+
+/**
+ * Initializes a player and returns it.
+ *
+ * @private
+ * @param  {Object} params
+ *         A parameters object. See README for details.
+ *
+ * @param  {Element} embed
+ *         An element that will be passed to the `bc()` function.
+ *
+ * @return {Object}
+ *         A Video.js player.
+ */
+const initPlayer = (params, embed) => {
+  const {playerId, embedId} = params;
+  const bc = window.bc[`${playerId}_${embedId}`] || window.bc;
+
+  if (!bc) {
+    return new Error(`missing bc function for ${playerId}`);
+  }
+
+  return bc(embed, params.options);
+};
+
+/**
+ * Loads a player from CDN and embeds it.
+ *
+ * @private
+ * @param  {Object} params
+ *         A parameters object. See README for details.
+ *
+ * @param  {Function} resolve
+ *         A function to call if a player is successfully initialized.
+ *
+ * @param  {Function} reject
+ *         A function to call if a player fails to be initialized.
+ */
+const loadPlayer = (params, resolve, reject) => {
+  checkParams(params);
+
+  const embed = createEmbed(params);
+
+  // If this is an iframe, all we need to do is create the embed code and
+  // inject it. Because there is no reliable way to hook into an iframe from
+  // the parent page, we simply resolve immediately upon creating the embed.
+  if (params.embedType === EMBED_TYPE_IFRAME) {
+    resolve(embed);
+    return;
+  }
+
+  const src = getUrl(params);
+
+  // If we've already downloaded this script, we should have the proper `bc`
+  // global and can bypass the script creation process.
+  if (downloadedScripts[src]) {
+    resolve(initPlayer(params, embed));
+    return;
+  }
+
```

Comment:
Do we want to cleanup any previous instances of the global `bc` object here?

Reply:
I think not. The player should be doing the right thing now w/r/t the `bc` global (i.e. overriding it, but keeping previous version(s) around).